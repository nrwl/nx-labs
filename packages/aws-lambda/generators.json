{
  "$schema": "http://json-schema.org/schema",
  "name": "aws-lamda",
  "version": "0.0.1",
  "generators": {
    "serverless": {
      "factory": "./src/generators/serverless/serverless",
      "schema": "./src/generators/serverless/schema.json",
      "description": "Creates a new project with AWS Lambda configuration"
    },
    "setup-serverless": {
      "factory": "./src/generators/setup-serverless/setup-serverless",
      "schema": "./src/generators/setup-serverless/schema.json",
      "description": "Adds AWS Lamda Serverless configuration to an existing project"
    },
    "preset": {
      "factory": "./src/generators/preset/preset",
      "schema": "./src/generators/preset/schema.json",
      "description": "Nx AWS Lambda Serverless preset generator",
      "hidden": true
    }
  }
}
