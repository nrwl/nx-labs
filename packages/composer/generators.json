{
  "generators": {
    "init": {
      "factory": "./src/generators/init/init",
      "schema": "./src/generators/init/schema.json",
      "description": "Initializes Composer in the current workspace",
      "hidden": true
    },
    "library": {
      "factory": "./src/generators/library/library",
      "schema": "./src/generators/library/schema.json",
      "description": "library generator"
    }
  }
}
