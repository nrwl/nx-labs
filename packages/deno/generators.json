{
  "$schema": "http://json-schema.org/schema",
  "name": "deno",
  "version": "0.0.1",
  "generators": {
    "application": {
      "factory": "./src/generators/application/generator",
      "schema": "./src/generators/application/schema.json",
      "description": "Create a new Deno application",
      "aliases": ["app"]
    },
    "library": {
      "factory": "./src/generators/library/library",
      "schema": "./src/generators/library/schema.json",
      "description": "Create a new Deno library",
      "aliases": ["lib"]
    },
    "init": {
      "factory": "./src/generators/init/generator",
      "schema": "./src/generators/init/schema.json",
      "description": "Intialize configurations for Deno"
    },
    "preset": {
      "factory": "./src/generators/preset/generator",
      "schema": "./src/generators/preset/schema.json",
      "description": "Create a new standalone Deno workspace with create-nx-workspace"
    },
    "add-import": {
      "aliases": ["import", "add"],
      "factory": "./src/generators/add-import/generator",
      "schema": "./src/generators/add-import/schema.json",
      "description": "Add an existing library to Deno imports"
    }
  }
}
