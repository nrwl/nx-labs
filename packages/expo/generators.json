{
  "name": "Nx Expo",
  "version": "0.1",
  "extends": ["@nrwl/workspace"],
  "schematics": {
    "init": {
      "factory": "./src/generators/init/init#expoInitSchematic",
      "schema": "./src/generators/init/schema.json",
      "description": "Initialize the @nrwl/expo plugin",
      "hidden": true
    },
    "application": {
      "factory": "./src/generators/application/application#expoApplicationSchematic",
      "schema": "./src/generators/application/schema.json",
      "aliases": ["app"],
      "x-type": "application",
      "description": "Create an application"
    },
    "library": {
      "factory": "./src/generators/library/library#expoLibrarySchematic",
      "schema": "./src/generators/library/schema.json",
      "aliases": ["lib"],
      "x-type": "library",
      "description": "Create a library"
    },
    "component": {
      "factory": "./src/generators/component/component#expoComponentSchematic",
      "schema": "./src/generators/component/schema.json",
      "description": "Create a component",
      "aliases": ["c"]
    }
  },
  "generators": {
    "init": {
      "factory": "./src/generators/init/init#expoInitGenerator",
      "schema": "./src/generators/init/schema.json",
      "description": "Initialize the @nrwl/expo plugin",
      "hidden": true
    },
    "application": {
      "factory": "./src/generators/application/application#expoApplicationGenerator",
      "schema": "./src/generators/application/schema.json",
      "aliases": ["app"],
      "x-type": "application",
      "description": "Create an application"
    },
    "library": {
      "factory": "./src/generators/library/library#expoLibraryGenerator",
      "schema": "./src/generators/library/schema.json",
      "aliases": ["lib"],
      "x-type": "library",
      "description": "Create a library"
    },
    "component": {
      "factory": "./src/generators/component/component#expoComponentGenerator",
      "schema": "./src/generators/component/schema.json",
      "description": "Create a component",
      "aliases": ["c"]
    }
  }
}
