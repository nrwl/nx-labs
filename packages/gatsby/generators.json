{
  "name": "nx/gatsby",
  "version": "0.1",
  "extends": ["@nrwl/react"],
  "schematics": {
    "init": {
      "factory": "./src/generators/init/init#gatsbyInitSchematic",
      "schema": "./src/generators/init/schema.json",
      "description": "Initialize the @nrwl/gatsby plugin",
      "hidden": true
    },
    "application": {
      "factory": "./src/generators/application/application#applicationSchematic",
      "schema": "./src/generators/application/schema.json",
      "aliases": ["app"],
      "x-type": "application",
      "description": "Create an application"
    },
    "component": {
      "factory": "./src/generators/component/component#componentSchematic",
      "schema": "./src/generators/component/schema.json",
      "description": "Create a component"
    },
    "page": {
      "factory": "./src/generators/page/page#pageSchematic",
      "schema": "./src/generators/page/schema.json",
      "description": "Create a page"
    }
  },
  "generators": {
    "init": {
      "factory": "./src/generators/init/init",
      "schema": "./src/generators/init/schema.json",
      "description": "Initialize the @nrwl/gatsby plugin",
      "hidden": true
    },
    "application": {
      "factory": "./src/generators/application/application",
      "schema": "./src/generators/application/schema.json",
      "aliases": ["app"],
      "x-type": "application",
      "description": "Create an application"
    },
    "component": {
      "factory": "./src/generators/component/component",
      "schema": "./src/generators/component/schema.json",
      "description": "Create a component"
    },
    "page": {
      "factory": "./src/generators/page/page",
      "schema": "./src/generators/page/schema.json",
      "description": "Create a page"
    }
  }
}
