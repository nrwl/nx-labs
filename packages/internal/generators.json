{
  "generators": {
    "migrate-to-nx-scope": {
      "factory": "./src/generators/migrate-to-nx-scope/generator",
      "schema": "./src/generators/migrate-to-nx-scope/schema.json",
      "description": "migrate-to-nx-scope generator"
    }
  }
}
