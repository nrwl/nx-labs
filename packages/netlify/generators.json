{
  "$schema": "http://json-schema.org/schema",
  "name": "netlify",
  "version": "0.0.1",
  "generators": {
    "serverless": {
      "factory": "./src/generators/serverless/serverless",
      "schema": "./src/generators/serverless/schema.json",
      "description": "Creates a new project with Netlify serverless configuration",
      "hidden": true
    },
    "setup-functions": {
      "factory": "./src/generators/setup-functions/setup-functions",
      "schema": "./src/generators/setup-functions/schema.json",
      "description": "Creates edge functions that can be deployed to Netlify to an existing project."
    },
    "preset": {
      "factory": "./src/generators/preset/preset",
      "schema": "./src/generators/preset/schema.json",
      "description": "Nx Netlify Serverless preset generator",
      "hidden": true
    }
  }
}
