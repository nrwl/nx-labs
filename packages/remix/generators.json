{
  "$schema": "http://json-schema.org/schema",
  "name": "NxRemix",
  "version": "0.0.1",
  "extends": ["@nrwl/react"],
  "generators": {
    "preset": {
      "implementation": "./src/generators/preset/preset.impl",
      "schema": "./src/generators/preset/schema.json",
      "description": "Generate a new Remix workspace",
      "hidden": true
    },
    "setup": {
      "implementation": "./src/generators/setup/setup.impl",
      "schema": "./src/generators/setup/schema.json",
      "description": "Setup a Remix in an existing workspace",
      "hidden": true
    },
    "application": {
      "implementation": "./src/generators/application/application.impl",
      "schema": "./src/generators/application/schema.json",
      "description": "Generate a new Remix application",
      "aliases": ["app"],
      "x-type": "application"
    },
    "library": {
      "implementation": "./src/generators/library/library.impl",
      "schema": "./src/generators/library/schema.json",
      "description": "Generate a new library",
      "aliases": ["lib"],
      "x-type": "library"
    },
    "route": {
      "implementation": "./src/generators/route/route.impl",
      "schema": "./src/generators/route/schema.json",
      "description": "Generate a new route"
    },
    "resource-route": {
      "implementation": "./src/generators/resource-route/resource-route.impl",
      "schema": "./src/generators/resource-route/schema.json",
      "description": "Generate a new resource route"
    },
    "action": {
      "implementation": "./src/generators/action/action.impl",
      "schema": "./src/generators/action/schema.json",
      "description": "Add an action function to an existing route"
    },
    "loader": {
      "implementation": "./src/generators/loader/loader.impl",
      "schema": "./src/generators/loader/schema.json",
      "description": "Add a loader function to an existing route"
    },
    "style": {
      "implementation": "./src/generators/style/style.impl",
      "schema": "./src/generators/style/schema.json",
      "description": "Generates a new stylesheet and adds it to an existing route"
    },
    "meta": {
      "implementation": "./src/generators/meta/meta.impl",
      "schema": "./src/generators/meta/schema.json",
      "description": "Add a meta function to an existing route"
    },
    "cypress": {
      "implementation": "./src/generators/cypress/cypress.impl",
      "schema": "./src/generators/cypress/schema.json",
      "description": "Generate a project for testing Remix apps using Cypress"
    }
  }
}
