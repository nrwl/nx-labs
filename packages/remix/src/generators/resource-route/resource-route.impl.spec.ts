import { Tree } from '@nx/devkit';
import { NameAndDirectoryFormat } from '@nx/devkit/src/generators/artifact-name-and-directory-utils';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { dirname } from 'path';
import applicationGenerator from '../application/application.impl';
import resourceRouteGenerator from './resource-route.impl';

describe('resource route', () => {
  let tree: Tree;

  beforeEach(async () => {
    tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    tree.write('.gitignore', `/node_modules/dist`);

    await applicationGenerator(tree, { name: 'demo' });
  });

  it('should not create a component', async () => {
    await resourceRouteGenerator(tree, {
      project: 'demo',
      path: '/example/',
      action: false,
      loader: true,
      skipChecks: false,
    });
    const fileContents = tree.read('apps/demo/app/routes/example.ts', 'utf-8');
    expect(fileContents).not.toMatch('export default function');
  });

  it('should throw an error if loader and action are both false', async () => {
    await expect(
      async () =>
        await resourceRouteGenerator(tree, {
          project: 'demo',
          path: 'example',
          action: false,
          loader: false,
          skipChecks: false,
        })
    ).rejects.toThrowErrorMatchingInlineSnapshot(
      `"The resource route generator requires either \`loader\` or \`action\` to be true"`
    );
  });

  describe.each([
    ['derived', 'apps/demo/app/routes/example.ts', 'demo'],
    ['derived', 'example', 'demo'],
    ['derived', 'example.ts', 'demo'],
    ['as-provided', 'apps/demo/app/routes/example', ''],
    ['as-provided', 'apps/demo/app/routes/example.ts', ''],
  ])(
    '--nameAndDirectoryFormat=%s',
    (
      nameAndDirectoryFormat: NameAndDirectoryFormat,
      path: string,
      project: string
    ) => {
      it(`should create correct file for path ${path}`, async () => {
        await resourceRouteGenerator(tree, {
          project,
          path,
          action: false,
          loader: true,
          skipChecks: false,
          nameAndDirectoryFormat,
        });

        expect(tree.exists('apps/demo/app/routes/example.ts')).toBeTruthy();
      });

      it('should error if it detects a possible missing route param because of un-escaped dollar sign', async () => {
        expect.assertions(3);

        await resourceRouteGenerator(tree, {
          project,
          path: `${dirname(path)}/route1/.ts`, // route.$withParams.tsx => route..tsx
          loader: true,
          action: true,
          skipChecks: false,
          nameAndDirectoryFormat,
        }).catch((e) => expect(e).toMatchSnapshot());

        await resourceRouteGenerator(tree, {
          project,
          path: `${dirname(path)}/route2//index.ts`, // route/$withParams/index.tsx => route//index.tsx
          loader: true,
          action: true,
          skipChecks: false,
          nameAndDirectoryFormat,
        }).catch((e) =>
          expect(e).toMatchInlineSnapshot(
            `[Error: Your route path has an indicator of an un-escaped dollar sign for a route param. If this was intended, include the --skipChecks flag.]`
          )
        );

        await resourceRouteGenerator(tree, {
          project,
          path: `${dirname(path)}/route3/.ts`, // route/$withParams.tsx => route/.tsx
          loader: true,
          action: true,
          skipChecks: false,
          nameAndDirectoryFormat,
        }).catch((e) => expect(e).toMatchSnapshot());
      });

      it(`should succeed if skipChecks flag is passed, and it detects a possible missing route param because of un-escaped dollar sign for ${path}`, async () => {
        const basePath =
          nameAndDirectoryFormat === 'as-provided'
            ? ''
            : 'apps/demo/app/routes';
        const normalizedPath = (
          dirname(path) === '' ? '' : `${dirname(path)}/`
        ).replace(basePath, '');
        await resourceRouteGenerator(tree, {
          project,
          path: `${normalizedPath}route1/..ts`, // route.$withParams.tsx => route..tsx
          loader: true,
          action: true,
          skipChecks: true,
          nameAndDirectoryFormat,
        });

        expect(tree.exists(`${basePath}/${normalizedPath}route1/..ts`)).toBe(
          true
        );

        await resourceRouteGenerator(tree, {
          project,
          path: `${normalizedPath}route2//index.ts`, // route/$withParams/index.tsx => route//index.tsx
          loader: true,
          action: true,
          skipChecks: true,
          nameAndDirectoryFormat,
        });

        expect(
          tree.exists(`${basePath}/${normalizedPath}route2/index.ts`)
        ).toBe(true);

        await resourceRouteGenerator(tree, {
          project,
          path: `${normalizedPath}route3/.ts`, // route/$withParams.tsx => route/.tsx
          loader: true,
          action: true,
          skipChecks: true,
          nameAndDirectoryFormat,
        });

        expect(tree.exists(`${basePath}/${normalizedPath}route3/.ts`)).toBe(
          true
        );
      });
    }
  );
});
