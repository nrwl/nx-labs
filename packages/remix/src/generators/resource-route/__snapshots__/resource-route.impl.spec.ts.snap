// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resource route --nameAndDirectoryFormat=as-provided should error if it detects a possible missing route param because of un-escaped dollar sign 1`] = `[Error: Your route path has an indicator of an un-escaped dollar sign for a route param. If this was intended, include the --skipChecks flag.]`;

exports[`resource route --nameAndDirectoryFormat=as-provided should error if it detects a possible missing route param because of un-escaped dollar sign 3`] = `[Error: Your route path has an indicator of an un-escaped dollar sign for a route param. If this was intended, include the --skipChecks flag.]`;

exports[`resource route --nameAndDirectoryFormat=as-provided should error if it detects a possible missing route param because of un-escaped dollar sign 4`] = `[Error: Your route path has an indicator of an un-escaped dollar sign for a route param. If this was intended, include the --skipChecks flag.]`;

exports[`resource route --nameAndDirectoryFormat=as-provided should error if it detects a possible missing route param because of un-escaped dollar sign 6`] = `[Error: Your route path has an indicator of an un-escaped dollar sign for a route param. If this was intended, include the --skipChecks flag.]`;

exports[`resource route --nameAndDirectoryFormat=derived should error if it detects a possible missing route param because of un-escaped dollar sign 1`] = `[Error: Your route path has an indicator of an un-escaped dollar sign for a route param. If this was intended, include the --skipChecks flag.]`;

exports[`resource route --nameAndDirectoryFormat=derived should error if it detects a possible missing route param because of un-escaped dollar sign 3`] = `[Error: Your route path has an indicator of an un-escaped dollar sign for a route param. If this was intended, include the --skipChecks flag.]`;

exports[`resource route --nameAndDirectoryFormat=derived should error if it detects a possible missing route param because of un-escaped dollar sign 4`] = `[Error: Your route path has an indicator of an un-escaped dollar sign for a route param. If this was intended, include the --skipChecks flag.]`;

exports[`resource route --nameAndDirectoryFormat=derived should error if it detects a possible missing route param because of un-escaped dollar sign 6`] = `[Error: Your route path has an indicator of an un-escaped dollar sign for a route param. If this was intended, include the --skipChecks flag.]`;

exports[`resource route --nameAndDirectoryFormat=derived should error if it detects a possible missing route param because of un-escaped dollar sign 7`] = `[Error: Your route path has an indicator of an un-escaped dollar sign for a route param. If this was intended, include the --skipChecks flag.]`;

exports[`resource route --nameAndDirectoryFormat=derived should error if it detects a possible missing route param because of un-escaped dollar sign 9`] = `[Error: Your route path has an indicator of an un-escaped dollar sign for a route param. If this was intended, include the --skipChecks flag.]`;
